feat: exchange connection

kev-hanwen-yang opened the pull request:
<!-- link to tickets -->
https://beequant.atlassian.net/jira/software/projects/CP/boards/1?selectedIssue=CP-131
---

## What happens here?
Added exchangeKey module
- Added create new exchange key api, including verifyExchangkey function and establishUserExchangeRealation function, both enabling store verified exchange key and relating exchange key with user and user’s exchange Added UserExchange module
- Add establistionRelations function to establish relationships between user, exchange and exchange key
- Add findUserExchange function to check any exchanges belonging to current user Added Exchange module
- Add createNewExchange and findExchangeById to create and find specific exchange

Resolve CP131
## How do I know this is working?
test in Postman with createNewExchange mutation
screenshot: 
1.add first api and secret key of biance
![image](https://github.com/user-attachments/assets/ecbbdbcc-7b89-4130-b58f-c311107d22a1)
Exchangekey Table
<img width="1323" alt="image" src="https://github.com/user-attachments/assets/9dc69b0e-a4b2-4091-8c8b-3e3694244338">
UserExchange Table
<img width="1313" alt="image" src="https://github.com/user-attachments/assets/9cb1754f-391e-428d-8163-c2fb0ae9040f">
Exchcange Table
<img width="1335" alt="image" src="https://github.com/user-attachments/assets/0f76db00-7809-4122-becd-9d524c7b401f">
2. add second api with same exchange(biance)
![image](https://github.com/user-attachments/assets/2c2d5e30-9557-4580-b087-c4dd558df6bf)
Exchangekey Table
<img width="1323" alt="image" src="https://github.com/user-attachments/assets/7d972737-1061-4965-a6ee-0c735915edec">
UserExchange Table
<img width="1326" alt="image" src="https://github.com/user-attachments/assets/9a57290a-b0e2-4f45-a557-34e1b8939891">
Exchcange Table
<img width="1325" alt="image" src="https://github.com/user-attachments/assets/9d61d7b1-be4a-4434-84bc-fcb8790fc9aa">
3. add existed api
![image](https://github.com/user-attachments/assets/917dacff-ce9b-4fcc-90f2-671bd6cb2221)
4. add invalied api
![image](https://github.com/user-attachments/assets/d3f32438-a375-4737-84ac-86e5ede12387)
5. add not supported exchange (currently only support biance)
![image](https://github.com/user-attachments/assets/7b180f4c-2754-4af5-9671-e677517c91f9)

Test Result:
<img width="713" alt="image" src="https://github.com/user-attachments/assets/55804d29-c8a9-4742-a1d0-ab8f90d9f369">


## Any notes?

<!--
  New PR Checklist

- Have you tag the right reviewer?
- Have you dm the reviewer?
- Maybe @ the reviewer in channel just to be safe

- is CI passing in your pr?
- Have you update the ticket with the right status?
-->
## Any notes?

<!--
  New PR Checklist

- Have you tag the right reviewer?
- Have you dm the reviewer?
- Maybe @ the reviewer in channel just to be safe

- is CI passing in your pr?
- Have you update the ticket with the right status?
-->

kev-hanwen-yang commented:
Added feature:

Implemented the ability to fetch all user connection balances for the authenticated user. This includes:

- Retrieving the balances of all user exchange connections.
- Filtering out zero balances and returning only the coins with non-zero balances.
- Returning the balances along with the corresponding user exchange connection details.

Chris-Wang commented on src/modules/user_exchange/user_exchange.service.ts:
```diff
@@ -0,0 +1,161 @@
+
+      return {
+        success: true,
+        message: 'Cool! Your exchange key is added successfully!',
```
the return message should be more subjective, since we're not using it directly to rending on client

Chris-Wang commented on src/modules/user_exchange/user_exchange.service.ts:
```diff
@@ -0,0 +1,161 @@
+      );
+
+      return {
+        success: true,
```
all return should come with `code` and `message`, we don't return `success`

Chris-Wang reviewed (changes requested):
- return message should be more subjective, it should only tell the state of the request
- all our return should follow `code`, `message`, `data` pattern, includes error return
- please merge the latest dev to your branch
- please fix lint issue:
![Uploading image.png…]()

Chris-Wang commented:
<img width="499" alt="image" src="https://github.com/BeeQuantAI/platform_api/assets/1610136/4bbfdb9f-a6e3-40cb-96eb-e175f54e5513">

- and squash commits

Chris-Wang commented on src/modules/user_exchange/user_exchange.resolver.ts:
```diff
@@ -0,0 +1,57 @@
+    return this.userExchangeService.getBalances(userId);
+  }
+
+  // TODO: template for exchangeManagement, updateExchange and deleteExchange tickets
```
commented code should not exist in code base unless it's necessary

Chris-Wang reviewed (changes requested):
Needs to:
- squash commits
- resolve previous comments
- we don't use `_` for folder or module name, consider renaming it to userExchange or user-exchange
- remove commented code

SeijaHuang committed:
Added exchangeKey module
- Added create new exchange key api, including verifyExchangkey function and establishUserExchangeRealation function, both enabling store verified exchange key and relating exchange key with user and user’s exchange
Added UserExchange module
- Add establistionRelations function to establish relationships between user, exchange and exchange key
- Add findUserExchange function to check any exchanges belonging to current user Added Exchange module
- Add createNewExchange and findExchangeById to create and find specific exchange
Resolve CP18

Chris-Wang reviewed (approved):
LGTM
